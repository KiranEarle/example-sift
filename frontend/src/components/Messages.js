import React, { Component } from 'react'
import MessageContent from './MessageContent'
import { condenseHeaderText } from '../helpers/condenseHeaderText'
import PropTypes from 'prop-types'
import Modal from 'react-modal'

Modal.setAppElement('#root')

class Messages extends Component {
  static Text = MessageContent

  constructor() {
    super()
    this.state = {
      modalIsOpen: false
    }
  }

  openModal = () => {
    this.setState({modalIsOpen: true});
  }

  closeModal = () => {
    this.setState({modalIsOpen: false});
  }


  renderHeader (message) {
    return (
      <div className='condensed-header'>
        <h2 className='message-header'>{condenseHeaderText(message)}</h2>
        <span className='expand-hide' onClick={() => this.openModal()}>More details</span>
      </div>
    )
  }


  render () {
    const { message } = this.props
    return (
      <div className='messages'>
        <Messages.Text header={this.renderHeader(message.body)} message={message} />
        <Modal
        isOpen={this.state.modalIsOpen}
        onRequestClose={this.closeModal}
        className='modal'
        overlayClassName='modal-overlay'>
          <Messages.Text header={message.body} message={message} fullDetails closeModal={this.closeModal} />
        </Modal>
      </div>
    )
  }
}

Messages.PropTypes = {
  /**
   * Takes an object with the message properties
   */
  message: PropTypes.object
}

export default Messages