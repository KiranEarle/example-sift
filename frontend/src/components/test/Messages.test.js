import React from 'react'
import Messages from '../Messages'

it('render with props without blowing up', () => {

  const message = {
    body: 'text',
    id: "text",
    subject: "more text",
    threadId: "even more text",
    wordCount: 198
  }

  const component = shallow(<Messages message={message} />);
  expect(component).toMatchSnapshot();
});