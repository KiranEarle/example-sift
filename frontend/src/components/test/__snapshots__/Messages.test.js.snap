// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render with props without blowing up 1`] = `
<div
  className="messages"
>
  <MessageContent
    header={
      <div
        className="condensed-header"
      >
        <h2
          className="message-header"
        >
          text...
        </h2>
        <span
          className="expand-hide"
          onClick={[Function]}
        >
          More details
        </span>
      </div>
    }
    message={
      Object {
        "body": "text",
        "id": "text",
        "subject": "more text",
        "threadId": "even more text",
        "wordCount": 198,
      }
    }
  />
  <Modal
    ariaHideApp={true}
    bodyOpenClassName="ReactModal__Body--open"
    className="modal"
    closeTimeoutMS={0}
    isOpen={false}
    onRequestClose={[Function]}
    overlayClassName="modal-overlay"
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    role="dialog"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={true}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
  >
    <MessageContent
      closeModal={[Function]}
      fullDetails={true}
      header="text"
      message={
        Object {
          "body": "text",
          "id": "text",
          "subject": "more text",
          "threadId": "even more text",
          "wordCount": 198,
        }
      }
    />
  </Modal>
</div>
`;
